spec/models/user_spec.rb

```diff
@@ -127,6 +127,58 @@
     end
   end
 
+  describe "#update" do
+    let(:existing_user) { create(:user) }
+
+    it "sets renamed_at if username changed" do
+      existing_user.update(login: "new_username")
+      expect(existing_user.renamed_at).to be_within(1.minute).of(Time.now.utc)
+    end
+
+    it "does not set renamed_at when username not changed" do
+      existing_user.update(email: "newemail@example.com")
+      expect(existing_user.renamed_at).to be_nil
+    end
+
+    context "username recently changed" do
+      before do
+        Timecop.freeze
+        existing_user.update(login: "#{existing_user.login}2")
```

Comment:
I'd prefer if the new values are simply typed out in full. They do not need to strictly be variations of the old value, so they do not need interpolation.